src/components/mixins/historyStats.ts

```diff
@@ -0,0 +1,122 @@
+import Vue from 'vue'
+import Component from 'vue-class-component'
+import {
+    HistoryStatsValueNames,
+    ServerHistoryStateAllPrintStatusEntry,
+    ServerHistoryStateJob,
+} from '@/store/server/history/types'
+import i18n from '@/plugins/i18n'
+
+@Component
+export default class HistoryStatsMixin extends Vue {
+    valueName!: HistoryStatsValueNames
+
+    get allPrintStatusChartData() {
+        return this.getChartData(this.$store.state.server.history.jobs ?? [])
+    }
+
+    get selectedPrintStatusChartData() {
+        return this.getChartData(this.$store.getters['server/history/getSelectedJobs'])
+    }
+
+    private getStatusColor(status: string) {
+        const colorMap: Record<string, string> = {
+            completed: '#BDBDBD',
+            in_progress: '#EEEEEE',
+            cancelled: '#616161',
+            default: '#424242',
+        }
+
+        return colorMap[status] ?? colorMap.default
+    }
+
+    private getLocalizedStatusName(status: string) {
+        return i18n.te(`History.StatusValues.${status}`, 'en')
+            ? i18n.t(`History.StatusValues.${status}`).toString()
+            : status
+    }
+
+    private getChartData(jobs: ServerHistoryStateJob[]) {
+        const output: ServerHistoryStateAllPrintStatusEntry[] = []
+        const hidePrintStatus = this.$store.state.gui.view.history.hidePrintStatus ?? []
+
+        jobs.forEach((current: ServerHistoryStateJob) => {
+            const index = output.findIndex((element) => element.name === current.status)
+            if (index !== -1) {
+                output[index].value += 1
+                output[index].valueFilament += current.filament_used
+                output[index].valueTime += current.print_duration
+                return
+            }
+
+            output.push({
+                name: current.status,
+                displayName: this.getLocalizedStatusName(current.status),
+                value: 1,
+                valueFilament: current.filament_used,
+                valueTime: current.print_duration,
+                itemStyle: {
+                    opacity: 0.9,
+                    color: this.getStatusColor(current.status),
+                    borderColor: '#1E1E1E',
+                    borderWidth: 2,
+                    borderRadius: 3,
+                },
+                showInTable: !hidePrintStatus.includes(current.status),
+            })
+        })
+
+        return output
+    }
+
+    private groupSmallEntries(
```

Comment:
I would not change this behavior, because if you select 3 jobs with different statuses in the list, you will also expect all 3 statuses to be displayed at the chart.

but i found another issue, when you just select 1 successfuly print and 1 failed print with 0 used filament. it will not display this "pies of cake" in the chart, but with the chart attribute "minAngle", it will always get displayed in the chart:
![image](https://github.com/user-attachments/assets/f2b69cb9-7383-46fa-a50f-a4f309459e30)

i pushed this change also, that you can test it with this fix.